### PR TITLE
Homebrew formula test block

### PR DESCRIPTION
**Pull request by Bl4Cc4t**
I rewrote the shell script with your example from https://github.com/Homebrew/homebrew-core/pull/46343#issuecomment-550053016 and added a generally better structure to it.

I also added Test::MockObject, which was reported missing when doing the tests.

Tests are working now (you can try it with [my latest commit](https://github.com/Bl4Cc4t/homebrew-other/commit/741fc5eeed8f88e1f6e53d0512a75100349d8105)), we now need 0.6.7 released before continuing over at [Homebrew-core](https://github.com/Homebrew/homebrew-core).
Should I close the PR and open a new one once we’re ready again?

### CODE REVIEW

File: tools/homebrew-start/lanraragi [hunk 1]
```diff
@@ -1,22 +1,35 @@
...
-  sed -i"" "s~<LRR_DATA_DIRECTORY>~${_LRR_DATA_DIRECTORY}~g" ${workdir}/redis.conf
+  sed -i"" "s~<LRR_DATA_DIRECTORY>~${LRR_DATA_DIRECTORY}~g" ${workdir}/redis.conf
 fi
-echo "Data Directory is set to '${_LRR_DATA_DIRECTORY}'"
 
-# create path if it does not exist yet
-mkdir -p "${_LRR_DATA_DIRECTORY}/log"
+# initialize LRR_LOG_DIRECTORY if not set
+if [[ ${LRR_LOG_DIRECTORY} == "" ]]; then
+  export LRR_LOG_DIRECTORY="${LRR_DATA_DIRECTORY}/log"
```
Comment (Difegue) [inline tools/homebrew-start/lanraragi]: This is a good opportunity to redirect logs to ~/Library/Logs for darwin:  
`_LRR_LOG_DIRECTORY="${HOME}/Logs/LANraragi"`

File: tools/homebrew-start/lanraragi [hunk 2]
```diff
@@ -1,22 +1,35 @@
...
-trap "kill 0" SIGINT
-
-# prefix/place where lrr is installed & modify redis config
+# prefix/place where lrr is installed
 workdir=$(dirname $0)/$(dirname $(readlink $0))/..
 
+# initialize LRR_DATA_DIRECTORY if not set
+if [[ ${LRR_DATA_DIRECTORY} == "" ]]; then
+  if [[ ${OSTYPE} == "darwin"* ]]; then
+    _LRR_DATA_DIRECTORY="${HOME}/Library/Application Support/LANraragi"
```
Comment (Difegue) [inline tools/homebrew-start/lanraragi]: If we have a custom LOG_DIR for darwin, it'd be better to move the `export` statement to a dedicated block in order to remove an indentation level here: 

```
if [[ ${OSTYPE} == "darwin"* ]]; then
  _LRR_DATA_DIRECTORY="${HOME}/Library/Application Support/LANraragi"
  _LRR_LOG_DIRECTORY="${HOME}/Logs/LANraragi"
else
  _LRR_DATA_DIRECTORY="${HOME}/LANraragi"
  _LRR_LOG_DIRECTORY="${HOME}/LANraragi/log"
fi

if [[ ${LRR_DATA_DIRECTORY} == "" ]]; then
  export LRR_DATA_DIRECTORY="${_LRR_DATA_DIRECTORY}"
fi

if [[ ${LRR_LOG_DIRECTORY} == "" ]]; then
  export LRR_LOG_DIRECTORY="${_LRR_LOG_DIRECTORY}"
fi
```

File: tools/homebrew-start/lanraragi [hunk 3]
```diff
@@ -1,22 +1,35 @@
...
+  fi
+  export LRR_DATA_DIRECTORY="${_LRR_DATA_DIRECTORY}"
+fi
+echo "Data Directory is set to '${LRR_DATA_DIRECTORY}'"
+
+# change values in redis.conf
 if [[ ${OSTYPE} == "darwin"* ]]; then
-  _LRR_DATA_DIRECTORY="${HOME}/Library/Application Support/LANraragi"
-  sed -i "" "s~<LRR_DATA_DIRECTORY>~${_LRR_DATA_DIRECTORY}~g" ${workdir}/redis.conf
+  sed -i "" "s~<LRR_DATA_DIRECTORY>~${LRR_DATA_DIRECTORY}~g" ${workdir}/redis.conf
```
Comment (Difegue) [inline tools/homebrew-start/lanraragi]:
```suggestion
  sed -i "" "s~<LRR_DATA_DIRECTORY>~${LRR_DATA_DIRECTORY}~g" ${workdir}/redis.conf
  sed -i "" "s~<LRR_LOG_DIRECTORY>~${LRR_LOG_DIRECTORY}~g" ${workdir}/redis.conf
```

same for the `-i""` variant for not-mac sed. The conf file also needs modification to introduce <LOG_DIRECTORY>.

**Comment by Difegue** [comment]
Yeah, I guess 0.6.6 as-is won't cut it due to the test problems.  
I try to put out a new release on a monthly basis, but if you want this in core before 0.6.7 I guess I could merge the log refactoring and this to master and slap a `0.6.6-homebrew` tag on it.

**Comment by Difegue** [comment]
Starting with commit 00359dda78311300ff662236000650557f53079a tests no longer require the redis server to be running as well, so you can simplify your homebrew formula too. :)

File: tools/homebrew-start/lanraragi (hunk 1 above)
Comment (Bl4Cc4t) [reply tools/homebrew-start/lanraragi]: The correct path is `"${HOME}/Library/Logs/LANraragi"`

File: tools/homebrew-start/lanraragi (hunk 2 above)
Comment (Bl4Cc4t) [reply tools/homebrew-start/lanraragi]: added!

File: tools/homebrew-start/lanraragi (hunk 3 above)
Comment (Bl4Cc4t) [reply tools/homebrew-start/lanraragi]: removed sed in favor of perl (I should’ve used perl in the first place lol)

**Comment by Bl4Cc4t** [comment]
No need to rush things, we can wait a bit for the 0.6.7 release!

File: tools/homebrew-start/lanraragi [hunk 4]
```diff
@@ -1,22 +1,40 @@
...
+fi
+if [[ ${LRR_LOG_DIRECTORY} == "" ]]; then
+  export LRR_LOG_DIRECTORY="${_LRR_LOG_DIRECTORY}"
 fi
-echo "Data Directory is set to '${_LRR_DATA_DIRECTORY}'"
 
-# create path if it does not exist yet
-mkdir -p "${_LRR_DATA_DIRECTORY}/log"
+# change values in redis.conf
+perl -pi.bak -e "s~<LRR_DATA_DIRECTORY>~${LRR_DATA_DIRECTORY}~g" ${workdir}/redis.conf
```
Comment (Difegue) [inline tools/homebrew-start/lanraragi]: What is the `-pi.bak` argument for here? afaik you can just run -e and it'll work as-is.

**Comment by Difegue** [comment]
Just got one question, but this looks good for a mergin'.

File: tools/homebrew-start/lanraragi (hunk 4 above)
Comment (Bl4Cc4t) [reply tools/homebrew-start/lanraragi]: We can leave the `.bak` part out (that just creates a backup file).
However, `-p` and `-i` are required.

Comment (Difegue) [reply tools/homebrew-start/lanraragi]: Right, re-read on my perl cli arguments and you're correct -- I don't use those too often :(

**Comment by Difegue** [comment]
Thanks!

**Pull request merged**